File: src/leetcode/2025/12/Solution392.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution392 {
public:
    bool isSubsequence(string s, string t) {
        int sLen = s.length();
        int tLen = t.length();
        int sIndex = 0;
        int tIndex = 0;

        while (sIndex < sLen && tIndex < tLen) {
            if (s[sIndex] == t[tIndex]) {
                sIndex++;
            }
            tIndex++;
        }

        return sIndex == sLen;
    }
};

int main() {
    Solution392 solution;

    string s1 = "abc";
    string t1 = "ahbgdc";
    bool result1 = solution.isSubsequence(s1, t1);
    // Expected output: true
    cout << (result1 ? "true" : "false") << endl;

    string s2 = "axc";
    string t2 = "ahbgdc";
    bool result2 = solution.isSubsequence(s2, t2);
    // Expected output: false
    cout << (result2 ? "true" : "false") << endl;
    return 0;
}

File: src/leetcode/2025/12/Solution392.java
public class Solution392 {
    public boolean isSubsequence(String s, String t) {
        int sIndex = 0, tIndex = 0;
        while (sIndex < s.length() && tIndex < t.length()) {
            if (s.charAt(sIndex) == t.charAt(tIndex)) {
                sIndex++;
            }
            tIndex++;
        }
        return sIndex == s.length();
    }

    public static void main(String[] args) {
        Solution392 solution = new Solution392();

        String s1 = "abc";
        String t1 = "ahbgdc";
        boolean result1 = solution.isSubsequence(s1, t1);
        System.out.println("Is \"" + s1 + "\" a subsequence of \"" + t1 + "\"? " + result1); // Expected output: true

        String s2 = "axc";
        String t2 = "ahbgdc";
        boolean result2 = solution.isSubsequence(s2, t2);
        System.out.println("Is \"" + s2 + "\" a subsequence of \"" + t2 + "\"? " + result2); // Expected output: false
    }
}
